occurrences/signals.py

```diff
@@ -43,3 +78,40 @@ def remove_message_id(sender, message, **kwargs):
     # ESP in order to avoid the message being identified as spam by some strict
     # filters.
     message.extra_headers.pop("Message-ID", None)
+
+
+@receiver([m2m_changed], sender=Occurrence.languages.through)
+@transaction.atomic
+def update_event_languages(sender, instance, action, pk_set, **kwargs):
+    """
+    When occurrence languages updates,
+    the languages should be synced to LinkedEvents Event languages.
+
+    NOTE: This will be called once by remove actions and once by additions.
+    """
+    # When languages are cleared, an empty list can be sent
+    if action == "post_clear":
+        send_event_languages_update(instance.p_event, [])
```

Comment:
Hmn, not sure if I get it right, does this mean if single occurrence languages are cleared, event language will become empty? If so it doesn't seem right because other occurrences can still have some languages left. 
In addition, an event cannot have empty `in_language` so this will also raise error when calling LE

Reply:
Nice catch! This was the action I didn't even test yet, because I thought it would be that simple :D. I need to add tests for all the 3 actions.

Reply:
Using event_language_ids instead, as is used always with post_add.